File: src/leibniz.java
import java.util.Scanner;
import java.io.File;
import java.io.FileNotFoundException;

public class leibniz {
    public static void main(String[] args) throws FileNotFoundException {
        Scanner s = new Scanner(new File("rounds.txt"));
        long rounds = s.nextLong();
        s.close();

        double sum = 0.0;
        double flip = -1.0;
        for (long i = 1; i <= rounds; i++) {
            flip *= -1.0;
            sum += flip / (2 * i - 1);
        }

        System.out.println(sum * 4.0);
    }
}

File: src/leibniz.pas
program leibniz;

var
  f: text;
  line: string;
  rounds, i: longint;
  pi, x: double;

begin
  assign(f, 'rounds.txt');
  reset(f);
  readln(f, line);
  close(f);

  val(line, rounds);

  pi := 1.0;
  x := 1.0;

  for i := 2 to rounds + 1 do
  begin
    x := x * -1;
    pi := pi + x / (2 * i - 1);
  end;

  pi := pi * 4;
  writeln(pi);
end.
